subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -0,0 +1,1321 @@
+ */
+
+/*
+ * Uses some original concepts by:
+ *         Joakim Eriksson <joakime@sics.se>
+ *         Niclas Finne <nfi@sics.se>
+ *         Joel Hoglund <joel@sics.se>
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_registry
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <fcntl.h>
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+
+#include "lwm2m_object.h"
+#include "lwm2m_engine.h"
+#include "lwm2m_util.h"
+#ifdef CONFIG_LWM2M_RD_CLIENT_SUPPORT
+#include "lwm2m_rd_client.h"
+#endif
+
+#define BINDING_OPT_MAX_LEN	3 /* "UQ" */
+#define QUEUE_OPT_MAX_LEN	2 /* "Q" */
+
+/* Resources */
+static sys_slist_t engine_obj_list;
+static sys_slist_t engine_obj_inst_list;
+
+/* Resource wrappers */
+sys_slist_t *lwm2m_engine_obj_list(void)
+{
+	return &engine_obj_list;
+}
+
+sys_slist_t *lwm2m_engine_obj_inst_list(void)
+{
+	return &engine_obj_inst_list;
+}
+
+/* Engine object */
+
+void lwm2m_register_obj(struct lwm2m_engine_obj *obj)
+{
+	sys_slist_append(&engine_obj_list, &obj->node);
+}
+
+void lwm2m_unregister_obj(struct lwm2m_engine_obj *obj)
+{
+	engine_remove_observer_by_id(obj->obj_id, -1);
+	sys_slist_find_and_remove(&engine_obj_list, &obj->node);
+}
+
+struct lwm2m_engine_obj *get_engine_obj(int obj_id)
+{
+	struct lwm2m_engine_obj *obj;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&engine_obj_list, obj, node) {
+		if (obj->obj_id == obj_id) {
+			return obj;
+		}
+	}
+
+	return NULL;
+}
+
+struct lwm2m_engine_obj_field *lwm2m_get_engine_obj_field(struct lwm2m_engine_obj *obj,
+							  int res_id)
+{
+	int i;
+
+	if (obj && obj->fields && obj->field_count > 0) {
+		for (i = 0; i < obj->field_count; i++) {
+			if (obj->fields[i].res_id == res_id) {
+				return &obj->fields[i];
+			}
+		}
+	}
+
+	return NULL;
+}
+
+struct lwm2m_engine_obj *lwm2m_engine_get_obj(
+					const struct lwm2m_obj_path *path)
+{
+	if (path->level < LWM2M_PATH_LEVEL_OBJECT) {
+		return NULL;
+	}
+
+	return get_engine_obj(path->obj_id);
+}
+/* Engine object instance */
+
+static void engine_register_obj_inst(struct lwm2m_engine_obj_inst *obj_inst)
+{
+	sys_slist_append(&engine_obj_inst_list, &obj_inst->node);
+}
+
+static void engine_unregister_obj_inst(struct lwm2m_engine_obj_inst *obj_inst)
+{
+	engine_remove_observer_by_id(
+			obj_inst->obj->obj_id, obj_inst->obj_inst_id);
+	sys_slist_find_and_remove(&engine_obj_inst_list, &obj_inst->node);
+}
+
+struct lwm2m_engine_obj_inst *get_engine_obj_inst(int obj_id,
+							 int obj_inst_id)
+{
+	struct lwm2m_engine_obj_inst *obj_inst;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&engine_obj_inst_list, obj_inst,
+				     node) {
+		if (obj_inst->obj->obj_id == obj_id &&
+		    obj_inst->obj_inst_id == obj_inst_id) {
+			return obj_inst;
+		}
+	}
+
+	return NULL;
+}
+struct lwm2m_engine_obj_inst *
+next_engine_obj_inst(int obj_id, int obj_inst_id)
```

Comment:
This line break seems weird - perhaps given the occasion try running `git-clang-format` to let it do the work?